lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -117,6 +117,24 @@ namespace Js
         return NewWithBufferT<LiteralString, true>(content, cchUseLength, scriptContext);
     }
 
+    JavascriptString* JavascriptString::NewWithByteCodeBuffer(__in_ecount(cchUseLength) const char16* content, charcount_t cchUseLength, Utf8SourceInfo* utf8SourceInfo, ScriptContext* scriptContext)
+    {
+        AssertMsg(content != nullptr, "NULL value passed to JavascriptString::New");
+        AssertMsg(IsValidCharCount(cchUseLength), "String length will overflow an int");
+
+        if (cchUseLength == 0)
```

Comment:
In a lot of places we also special-case length 1. It's technically allowed to have single-character strings that aren't of type SingleCharString, but we get the most benefit from SingleCharString if we use it consistently. Check out JavascriptString::NewWithBufferT for an example.